packages/@react-aria/menu/src/useMenu.ts

```diff
@@ -72,11 +75,31 @@ export function useMenu<T>(props: AriaMenuOptions<T>, state: TreeState<T>, ref:
     onAction: props.onAction
   });
 
+  let {keyboardProps} = useKeyboard({
+    onKeyDown(e) {
+      // TODO: Let Tab propagate so FocusScope handle it. Revisit if we decide to close all menus
+      // Also let Escape propagate so useOverlay handles closing the Menu overlay
+      if (e.key === 'Tab' || e.key === 'Escape') {
+        e.continuePropagation();
+      }
+
+      onKeyDown && onKeyDown(e);
+    },
+    onKeyUp(e) {
+      // Need to continue keyup propagation so MenuTrigger button's press state is properly reset, otherwise it gets stuck in pressed state
+      e.continuePropagation();
+      onKeyUp && onKeyUp(e);
+    }
+  });
```

Comment:
`useKeyboard` isn't strictly required persay, was just something we decided to add to ensure most other keydown events had `stopPropagation` by default. Happy to get rid of it and handle the stopPropagation in useSubMenuTrigger's `submenuKeyDown` so that we only stop propagation for keyboard events that we specifically handle